SDL/Segment.cu

```diff
@@ -918,27 +922,68 @@ __global__ void SDL::createSegmentsInGPUv2(struct SDL::modules& modulesInGPU, st
     int blockxSize = blockDim.x*gridDim.x;
     int blockySize = blockDim.y*gridDim.y;
     int blockzSize = blockDim.z*gridDim.z;
+
+    __shared__ short subdets[MAX_CONNECTED_MODULES], layers[MAX_CONNECTED_MODULES], sides[MAX_CONNECTED_MODULES], rods[MAX_CONNECTED_MODULES], rings[MAX_CONNECTED_MODULES], innerSubdet, innerLayer, innerSide, innerRod, innerRing;
+    __shared__ float drdzs[MAX_CONNECTED_MODULES], innerDrdz;
+    __shared__ ModuleType moduleType[MAX_CONNECTED_MODULES], innerModuleType;
+
     for(uint16_t innerLowerModuleIndex = blockIdx.z * blockDim.z + threadIdx.z; innerLowerModuleIndex< (*modulesInGPU.nLowerModules); innerLowerModuleIndex += blockzSize){
 
+    unsigned int nInnerMDs = mdsInGPU.nMDs[innerLowerModuleIndex];
+    if(nInnerMDs == 0) continue;
+    unsigned int innerMDPrefix = rangesInGPU.miniDoubletModuleIndices[innerLowerModuleIndex];
+
     unsigned int nConnectedModules = modulesInGPU.nConnectedModules[innerLowerModuleIndex];
+    if(nConnectedModules == 0) continue;
+    //shared memory stuff
+    if(threadIdx.x == 0 and threadIdx.z == 0)
+    {
+        for(unsigned int i = threadIdx.y; i < nConnectedModules; i+=blockDim.y)
+        {
+            subdets[i] = modulesInGPU.subdets[modulesInGPU.moduleMap[innerLowerModuleIndex * MAX_CONNECTED_MODULES + i]];
+            layers[i] = modulesInGPU.layers[modulesInGPU.moduleMap[innerLowerModuleIndex * MAX_CONNECTED_MODULES + i]];
```

Comment:
just in case the compiler is not optimizing away the inner index computation, does it get any faster with a temp
```suggestion
            auto moduleIdx = modulesInGPU.moduleMap[innerLowerModuleIndex * MAX_CONNECTED_MODULES + i];
            subdets[i] = modulesInGPU.subdets[moduleIdx];
            layers[i] = modulesInGPU.layers[moduleIdx];
```
... (similar to the remaining lines)